server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1749,7 +1749,7 @@ int sizeInBytes() {
     public void finalizeSnapshot(final FinalizeSnapshotContext finalizeSnapshotContext) {
         assert ThreadPool.assertCurrentThreadPool(ThreadPool.Names.SNAPSHOT);
         final long repositoryStateId = finalizeSnapshotContext.repositoryStateId();
-        final ShardGenerations shardGenerations = finalizeSnapshotContext.updatedShardGenerations();
+        final ShardGenerations shardGenerations = finalizeSnapshotContext.updatedShardGenerations().liveIndices();
```

Comment:
Nit: maybe inline this, it's only used in one place, and we now need to care about _which_ `ShardGenerations` we're talking about so the variable name is ambiguous.

Reply:
Good point. See 1071d952abb164fba23fd0063c6d2ef74fea56cc